core/src/main/scala/org/apache/spark/shuffle/FileShuffleBlockResolver.scala

```diff
@@ -105,7 +105,8 @@ private[spark] class FileShuffleBlockResolver(conf: SparkConf)
    * when the writers are closed successfully
    */
   def forMapTask(shuffleId: Int, mapId: Int, numBuckets: Int, serializer: Serializer,
-      writeMetrics: ShuffleWriteMetrics): ShuffleWriterGroup = {
+      writeMetrics: ShuffleWriteMetrics,
+      getDiskWriter: (BlockId, File, SerializerInstance, Int, ShuffleWriteMetrics) => BlockObjectWriter = blockManager.getDiskWriter): ShuffleWriterGroup = {
```

Comment:
Thanks for pointing that out, @JoshRosen. I'll update this to be more Java-friendly tomorrow. In the meantime, feel free to make suggestions on how to customize the creation of a BlockObjectWriter when calling forMapTask().